control-plane/connect-inject/controllers/pod/pod_controller_test.go

```diff
@@ -1015,17 +1620,76 @@ func expectedWorkloadMatches(t *testing.T, client pbresource.ResourceServiceClie
 	require.True(t, proto.Equal(actualWorkload, expectedWorkload))
 }
 
-func loadResource(t *testing.T, client pbresource.ResourceServiceClient, id *pbresource.ID, proto proto.Message) {
-	if id == nil || proto == nil {
+func expectedHealthStatusMatches(t *testing.T, client pbresource.ResourceServiceClient, name string, expectedHealthStatus *pbcatalog.HealthStatus) {
+	req := &pbresource.ReadRequest{Id: getHealthStatusID(name, metav1.NamespaceDefault, constants.DefaultConsulPartition)}
+
+	res, err := client.Read(context.Background(), req)
+
+	if expectedHealthStatus == nil {
+		require.Error(t, err)
+		s, ok := status.FromError(err)
+		require.True(t, ok)
+		require.Equal(t, codes.NotFound, s.Code())
+		return
+	}
+
+	require.NoError(t, err)
+	require.NotNil(t, res)
+
+	require.Equal(t, name, res.GetResource().GetId().GetName())
+	require.Equal(t, constants.DefaultConsulNS, res.GetResource().GetId().GetTenancy().GetNamespace())
+	require.Equal(t, constants.DefaultConsulPartition, res.GetResource().GetId().GetTenancy().GetPartition())
+
+	require.NotNil(t, res.GetResource().GetData())
+
+	actualWorkload := &pbcatalog.HealthStatus{}
+	err = res.GetResource().GetData().UnmarshalTo(actualWorkload)
+	require.NoError(t, err)
+
+	require.True(t, proto.Equal(actualWorkload, expectedHealthStatus))
+}
+
+func expectedProxyConfigurationMatches(t *testing.T, client pbresource.ResourceServiceClient, name string, expectedProxyConfiguration *pbmesh.ProxyConfiguration) {
```

Comment:
~~ I feel like maybe we could convert these to a generic "resource compare" fn, but no strong feeling about doing it now

Reply:
I like this idea and started playing with it. Implementing it is a little messier than I'd like, so I'm thinking I'll revisit it as part of the ENT tests.